cmi.proto

```diff
@@ -7,236 +7,273 @@ import "google/protobuf/wrappers.proto";
+    // This field is OPTIONAL.
+    repeated PluginCapability capabilities = 1;
 }
 
-message GetMachineResponse {
-  // Exists tells if the machine exists on the cloud provider
-  // This field is REQUIRED.
-  bool Exists = 1;
-
-  // Status contains the possible status for machines running on the cloud
-  enum Status {
-    Unknown = 0;
-    Running = 1;
-    Stopped = 2;
-  }
-  Status MachineStatus = 2;
+// Specifies a capability of the CMI-Plugin.
+message PluginCapability {
+    message RPC {
+        enum Type {
+            // UNKNOWN is used to specific an capaability beyond the set
+            // provided below
+            UNKNOWN = 0;
+
+            // CREATE_MACHINE tells that the plugin implements the
+            // CreateMachine() interface
+            CREATE_MACHINE = 1;
+
+            // DELETE_MACHINE tells that the plugin implements the
+            // DeleteMachine() interface
+            DELETE_MACHINE = 2;
+
+            // GET_MACHINE tells that the plugin implements the
+            // GetMachine() interface
+            GET_MACHINE = 3;
+
+            // SHUTDOWN_MACHINE tells that he plugin implements the
+            // ShutDownMachine() interface
+            SHUTDOWN_MACHINE = 4;
+
+            // GET_LIST_OF_VOLUMEIDS_FOR_EXISTING_PVS tells if the plugin
+            // implements the GetListOfVolumeIDsForExistingPVs() interface
+            GET_LIST_OF_VOLUMEIDS_FOR_EXISTING_PVS = 5;
+        }
+        Type type = 1;
+    }
+    oneof type {
+        // RPC that the plugin supports.
+        RPC rpc = 1;
+    }
 }
 
-message ListMachinesRequest {
-  // ProviderSpec is the configuration needed to list machines.
-  // Driver should parse this raw data into pre-defined spec in their respective projects.
-  // This field is REQUIRED.
-  bytes ProviderSpec = 1;
-
-  // Secrets is the map containing necessary credentials for cloud-provider to list the machines.
-  // This field is OPTIONAL.
-  map<string, bytes> Secrets = 2 [(cmi_secret) = true];
+message ProbeRequest {
+    // Intentionally empty.
 }
 
-message ListMachinesResponse {
-  // MachineList is the map of list of machines. Format for the map should be map<MachineID, MachineName>.
-  // This field is REQUIRED.
-  map<string, string> MachineList = 1;
+message ProbeResponse {
+    // Readiness allows a plugin to report its initialization status back
+    // to the CMI-Client. Initialization for some plugins MAY be time consuming
+    // and it is important for a CMI-Client to distinguish between the following
+    // cases:
+    //
+    // 1) The plugin is in an unhealthy state and MAY need restarting. In
+    //    this case a gRPC error code SHALL be returned.
+    // 2) The plugin is still initializing, but is otherwise perfectly
+    //    healthy. In this case a successful response SHALL be returned
+    //    with a readiness value of `false`. Calls to the plugin's
+    //    Controller and/or Node services MAY fail due to an incomplete
```

Comment:
`Controller and/or Node services` should be replaced with `Machine service`.
CSI has these 2 services, however, CMI has only Machine service.

Reply:
Done.